scripts/test.py

```diff
@@ -0,0 +1,146 @@
+    build_script = importlib.import_module('build')
+    return build_script
+
+
+def parse_args(build_script):
+    '''
+    Parse command line arguments
+    '''
+    arg_parser = argparse.ArgumentParser(
+        description="gfxreconstruct build script",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    arg_parser.add_argument('--version', dest='version',
+                            action='version', version=str(build_script.VERSION))
+    arg_parser.add_argument(
+        '--test-exe', dest='test_exe', metavar='TEST_EXE',
+        action='store', default=None,
+        help='Test executable to run')
+    arg_parser.add_argument(
+        '--test-args', dest='test_args', metavar='TEST_ARGS', nargs='+',
+        action='store',
+        help='Test arguments passed to the test executable')
+    arg_parser.add_argument(
+        '--build-dir', dest='build_dir', metavar='BUILD_DIR',
+        action='store', default=None)
+    arg_parser.add_argument(
+        '-a', '--arch', dest='architecture',
+        metavar='ARCH', action='store', choices=build_script.ARCHITECTURES,
+        default=build_script.DEFAULT_ARCHITECTURE,
+        help='Build target architecture. Can be one of: {0}'.format(
+                ', '.join(build_script.ARCHITECTURES)))
+    arg_parser.add_argument(
+        '-c', '--config', dest='configuration',
+        metavar='CONFIG', action='store', choices=build_script.CONFIGURATIONS,
+        default=build_script.DEFAULT_CONFIGURATION,
+        help='Build target configuration. Can be one of: {0}'.format(
+            ', '.join(build_script.CONFIGURATIONS)))
+    return arg_parser.parse_args()
+
+
+def run_test(test_exe, test_args):
+    '''
+    Run a single test
+    '''
+    run_test_args = [test_exe]
+    if args.test_args is not None:
+        run_test_args.extend(args.test_args)
+    run_test_work_dir = os.path.split(test_exe)[0]
+    if '' == run_test_work_dir:
+        run_test_work_dir = os.getcwd()
+    try:
+        test_result = subprocess.run(run_test_args, cwd=run_test_work_dir)
+        if 0 != test_result.returncode:
+            raise TestError('Running test', ' '.join(run_test_args), 'failed')
+    except Exception:
+        raise TestError('Failed to run test', test_exe, ' '.join(test_args))
+
+
+# Main entry point
+if '__main__' == __name__:
+    try:
+        build_script = import_build_script()
+        args = parse_args(build_script)
+        tests = []
+        if args.test_exe is None:
+            for test in ALL_TESTS.items():
+                test_exe_dir = args.build_dir
+                if test_exe_dir is None:
+                    test_exe_dir = os.path.join(
+                        build_script.BUILD_CONFIGS[args.configuration],
+                        platform.system().lower(),
+                        args.architecture,
+                        'bin')
+                test_exe = os.path.join(test_exe_dir, test[0])
+                test_args = copy.deepcopy(test[1])
```

Comment:
Why `copy.deepcopy()`?  I can't think of any reason it would be necessary.

Reply:
I don't want to modify the original test arguments, since they might be used later.